### ISSUE TITLE
Improve eclipse plugin handling of test source sets

### ISSUE THREAD

**Issue by jjstreet** [gradle/gradle#1238]
When using the standard source sets for Java, the eclipse and eclipse-wtp plugins should make sure that the output folders for test source folders are different to prevent test classes from being included in Eclipse-built war and ear files.

### Expected Behavior
The eclipse plugin will set a different output folder on the class path for test source sets. The eclipse-wtp will not add test source sets as WbResources to the WtpComponent file

### Current Behavior
Right now `src/test` and `src/main` both have `bin` as the standard output folder for eclipse. While eclipse-wtp doesn't add `src/test` to the deployment assembly information, the output folder still contains compiled test classes. Changing the output folder for `src/test` causes eclipse-wtp plugin to add it as a distinct item in the deployment assembly.

### Context
Ensuring test sources don't get added to eclipse built artifacts improves eclipse integration.

### Current Workaround
Below is how the above can be achieved currently
```
import org.gradle.plugins.ide.eclipse.model.WbResource

eclipse {
    classpath.file.whenMerged {
        entries.findAll {
            it.path.startsWith('src/test')
        }*.output = 'bin-test'
    }
    
    wtp.component.file.whenMerged {
        wbModuleEntries.removeAll {
            it instanceof WbResource && it.sourcePath.startsWith('/src/test')
        }
    }
}
```
### Your Environment
Gradle 3.3

**Comment by jjstreet** [comment]
I spent some time looking at the sources for the `eclipse` plugin and I'm not sure how this could be added. Are there any resources that talk about how this plugin works?

**Comment by Adrodoc55** [comment]
This seems to be fixed in Gradle 4.4 as the default output dir for the main SourceSet is now "bin/main" and for the test SourceSet it's "bin/test". This change isn't even mentioned in the release notes, but the change of the default project output is listed: https://docs.gradle.org/4.4/release-notes.html#changes-in-the-eclipse-plugin
